calculator/main.py

```diff
@@ -37,14 +61,21 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    
+    #calc['operations'] = dict(list(calc['operations'].items()) + list(operation.items()))
+    
+    newname = list(operation.keys())[0]
```

Comment:
You could use `isinstance()` here to make sure `operation` is of the type that you want.